app/models/course.rb

```diff
@@ -9,6 +9,8 @@ class Course < ApplicationRecord
 
   audited associated_with: :provider
 
+  before_save :set_opened_on_apply_at
```

Comment:
Historically we haven't handled this kind of thing with ActiveRecord callbacks and I think that's generally been good. Would it clearer to set the date in `OpenProviderCourses` — or at least to have an `open!` method on each course?

And do we need to set back to `nil`? It's still fact that the course was opened on apply at that date.

Reply:
+1 for `#open!`
Regarding `nil`,  in our last conversation as @mnacos pointed out, we were hoping to only have it set if the value is true. Since there seems to precedence with timestamps such as `rejected_at` never being unset maybe we can keep it if a course is reverted back to false, but always update it if a value moves from false to true.

Reply:
I've added an `#open!` method. May result in more saves during a sync, but we're planning to drop audits for courses anyway.